unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/JoinChannelHUD/Tests/JoinChannelComponentControllerShould.cs

```diff
@@ -0,0 +1,74 @@
+using DCL;
+using NSubstitute;
+using NSubstitute.Extensions;
+using NUnit.Framework;
+
+public class JoinChannelComponentControllerShould
+{
+    private JoinChannelComponentController joinChannelComponentController;
+    private IJoinChannelComponentView joinChannelComponentView;
+    private IChatController chatController;
+    private DataStore_Channels channelsDataStore;
+
+    [SetUp]
+    public void SetUp()
+    {
+        joinChannelComponentView = Substitute.For<IJoinChannelComponentView>();
+        chatController = Substitute.For<IChatController>();
+        channelsDataStore = new DataStore_Channels();
+        joinChannelComponentController = Substitute.ForPartsOf<JoinChannelComponentController>();
+        joinChannelComponentController.Configure().CreateJoinChannelView().Returns(info => joinChannelComponentView);
+        joinChannelComponentController.Initialize(chatController, channelsDataStore);
+    }
+
+    [TearDown]
+    public void TearDown() { joinChannelComponentController.Dispose(); }
+
+    [Test]
+    public void InitializeCorrectly()
+    {
+        // Assert
+        Assert.AreEqual(joinChannelComponentView, joinChannelComponentController.joinChannelView);
+        Assert.AreEqual(chatController, joinChannelComponentController.chatController);
+        Assert.AreEqual(channelsDataStore, joinChannelComponentController.channelsDataStore);
+    }
+
+    [Test]
+    [TestCase("TestId")]
+    [TestCase(null)]
+    public void RaiseOnChannelToJoinChangedCorrectly(string testChannelId)
+    {
+        // Act
+        joinChannelComponentController.OnChannelToJoinChanged(testChannelId, null);
```

Comment:
I'm not totally sure about that... what we're doing here are **unit tests** for each independent method in the class, so in this case I want to test the logic of the `OnChannelToJoinChanged` method. Who calls this method (in this case an external event) is part of other tests in my opinion.

Anyway, I have changed the tests in order to raise the different events instead of the moethods directly.

Done in https://github.com/decentraland/unity-renderer/pull/2724/commits/c00ce1175988e60ba915b136c140286276bc8282